lib/collection/src/lib.rs

```diff
@@ -447,14 +463,26 @@ impl Collection {
         self.shards.values()
     }
 
+    /// Handle collection updates from peers.
+    ///
+    /// Shard transfer aware.
     pub async fn update_from_peer(
         &self,
         operation: CollectionUpdateOperations,
         shard_selection: ShardId,
         wait: bool,
     ) -> CollectionResult<UpdateResult> {
-        let local_shard = self.local_shard_by_id(shard_selection)?;
-        local_shard.get().update(operation.clone(), wait).await
+        let target_shard = match self.local_shard_by_id(shard_selection) {
+            Ok(local_shard) => local_shard.get(),
+            Err(err) => {
+                // verify if maybe it targets a shard transfer
+                match self.transfer_shard_by_id(shard_selection) {
+                    None => return Err(err),
+                    Some(shard_transfer) => Arc::new(shard_transfer),
+                }
+            }
+        };
```

Comment:
If we want to handle it like that, I think it is better for `local_shard_by_id` to return `Option` as it is not strictly an error